Source/Core/Common/x64Emitter.cpp

```diff
@@ -2426,7 +2426,7 @@ void XEmitter::MOVDDUP(X64Reg regOp, const OpArg& arg)
   else
   {
     if (!arg.IsSimpleReg(regOp))
-      MOVSD(regOp, arg);
+      MOVAPS(regOp, arg);
```

Comment:
I think this may be reachable when `arg` is a memory location. In that case MOVAPS will read 128 bits instead of 64, which seems like a bad idea. Perhaps we could do something like this:

```
if (!arg.isSimpleReg())
  MOVSD(regOp, arg);
else if (regOp != arg.getSimpleReg())
  MOVAPS(regOp, arg);
```

Fun fact: MOVSD actually _does_ zero the upper half of the destination register when used with a memory location. Yes, really.

Reply:
Sounds reasonable!